app/controllers/posts_controller.rb

```diff
@@ -7,4 +9,39 @@ def index
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = @current_user.posts.build
+  end
+
+  def create
+    @post = @current_user.posts.build(post_params)
+
+    if @post.save
+      redirect_to user_post_path(@current_user, @post), notice: 'Post was successfully created!'
+    else
+      render :new
+    end
+  end
```

Comment:
- [ ] Good job so far but there is a mix up I would recommend you quickly fix up, I explained it in [this video](https://www.loom.com/share/3f337b6f1de44a4c8b65f0d18c330b14?sid=f8bdb76c-29e3-4044-9bbe-f8947a198d5f), the issue is as regards the post, it goes to the last user as I showed in the video, whereas, it should be attached to the user that writes it, kindly take a second look at it.